### PR TITLE
terminal: fix terminal attribute overflow

### LINKED ISSUE #11548: Segment fault when using :terminal
I'm having segment fault errors after selecting some item with fzf. Has anyone experimented any problem with it too?

image

I don't know neither how to reproduce nor how to get some stack trace for debug as this error happens every now and then without any pattern, just after selecting some item from the fzf list. Can someone help me with that? 

My currently configs:
- `nvim --version`: v0.4.3
- `vim -u DEFAULTS` (version: ) behaves differently?
- Operating system/version: MacOS 10.15.2
- Terminal name/version: Iterm 3.3.7
- `$TERM`: xterm-256color
- Fzf.vim plugin from master

### PR DESCRIPTION
**Pull request by erw7**
fixes #11548

### CODE REVIEW

File: src/nvim/screen.c [hunk 1]
```diff
@@ -2804,8 +2804,8 @@ win_line (
     off += col;
   }
 
-  // wont highlight after 1024 columns
-  int term_attrs[1024] = {0};
+  // wont highlight after TERM_ATTRS_MAX(1024) columns
```
Comment (teto) [inline src/nvim/screen.c:2807]: remove the 1024, one thing less to update.

File: src/nvim/screen.c [hunk 2]
```diff
@@ -4172,7 +4172,7 @@ win_line (
         int n = wp->w_p_rl ? -1 : 1;
         while (col >= 0 && col < grid->Columns) {
           schar_from_ascii(linebuf_char[off], ' ');
-          linebuf_attr[off] = term_attrs[vcol];
+          linebuf_attr[off] = vcol > TERM_ATTRS_MAX ? 0 : term_attrs[vcol];
```
Comment (teto) [inline src/nvim/screen.c:4175]: shouldn't it be `>= ` ?

**Review by teto — APPROVED** [review]
feel free to merge (after CI)

File: src/nvim/screen.c [hunk 3]
```diff
@@ -4172,7 +4172,7 @@ win_line (
         int n = wp->w_p_rl ? -1 : 1;
         while (col >= 0 && col < grid->Columns) {
           schar_from_ascii(linebuf_char[off], ' ');
-          linebuf_attr[off] = term_attrs[vcol];
+          linebuf_attr[off] = vcol >= TERM_ATTRS_MAX ? 0 : term_attrs[vcol];
```
Comment (justinmk) [inline src/nvim/screen.c:4175]: could use `ARRAY_SIZE(term_attrs)`

**Comment by justinmk** [comment]
backported https://github.com/neovim/neovim/pull/12822

**Pull request merged**